File: tests/fixtures/sample.lua
-- SPDX-License-Identifier: GPL-2.0-or-later

local DEFAULT_LIMIT = 42

---@class Worker
local Worker = {}
Worker.__index = Worker

function Worker.new(name)
  return setmetatable({ name = name, active = true }, Worker)
end

function Worker:run(items)
  local pattern = "^task:%d+$"
  for _, item in ipairs(items) do
    if self.active and item:match(pattern) then
      print(("%s:%d"):format(self.name, DEFAULT_LIMIT))
    end
  end
end

return Worker

File: tests/fixtures/sample.py
# SPDX-License-Identifier: GPL-2.0-or-later
import re
from dataclasses import dataclass

DEFAULT_LIMIT = 42


@dataclass
class Worker:
    name: str
    active: bool = True

    def run(self, items: list[str]) -> int:
        pattern = re.compile(r"^task:\d+$")
        matches = [item for item in items if self.active and pattern.match(item)]
        return min(len(matches), DEFAULT_LIMIT)


result = Worker("limei").run(["task:1", "skip"])
